terraform/workflow.yaml

```diff
@@ -0,0 +1,151 @@
+main:
+  params: [job]
+  steps:
+    - vars:
+        assign:
+          - project_id: ${sys.get_env("GOOGLE_CLOUD_PROJECT_ID")}
+          - collection: "test-runs"
+          - job_id: ${job.jobId}
+          - zone: "us-west1-b"
+          - region: "us-west1"
+          - instance_name: "test-runner"
+          - machine_type: "e2-small"
+          - project_number: ${sys.get_env("GOOGLE_CLOUD_PROJECT_NUMBER")}
+          - service_account_email: ${project_number + "-compute@developer.gserviceaccount.com"}
+
+    - create_job_in_firestore:
+        call: googleapis.firestore.v1.projects.databases.documents.createDocument
+        args:
+          collectionId: ${collection}
+          parent: ${"projects/" + project_id + "/databases/(default)/documents"}
+          query:
+            documentId: ${job_id}
+          body:
+            fields:
+              jobId:
+                stringValue: ${job_id}
+    - get_document:
+        call: googleapis.firestore.v1.projects.databases.documents.get
+        args:
+          name: ${"projects/" + project_id + "/databases/(default)/documents/" + collection + "/" + job_id}
+        result: got
+    - validate_saved_document:
+        switch:
+          - condition: ${got.fields.jobId.stringValue == job_id}
+            next: get_latest_image
+    - failed:
+        raise: ${"got unexpected document"}
+
+    #Get the latest boot image.
+    - get_latest_image:
+        call: googleapis.compute.v1.images.getFromFamily
+        args:
+          family: "cos-stable"
+          project: "cos-cloud"
+        result: getFromFamilyResult
+    #Create and start compute engine vm.
+    - insert_machine:
+        call: googleapis.compute.v1.instances.insert
+        args:
+          project: ${project_id}
+          zone: ${zone}
+          body:
+            tags:
+              items:
+                - http-server
+            name: ${instance_name}
+            machineType: ${"zones/" + zone + "/machineTypes/" + machine_type}
+            disks:
+              - initializeParams:
+                  sourceImage: ${"projects/cos-cloud/global/images/" + getFromFamilyResult.name}
+                boot: true
+                autoDelete: true
+            networkInterfaces:
+              - accessConfigs:
+                  - name: "External NAT"
+                    networkTier: "STANDARD"
+                subnetwork: ${"projects/" + project_id + "/regions/" + region +"/subnetworks/default"}
+            metadata:
+              items:
+                - key: "startup-script"
+                  value: "#! /bin/bash \ncd tmp\nmkdir test-runner\ncd test-runner\npwd\nls\ngit clone https://github.com/adarsh-jaiswal/graphql-java-test-runner.git\ncd graphql-java-test-runner/ \ngit checkout bcf3a7a \ndocker build --no-cache -t test-runner .\ndocker run --publish 80:8080 test-runner"
```

Comment:
This script is used to-
1. Checkout test-runner repo.
2. checkout a specific commit as the docker image finds the jar from build directory.(will be modified later).
3. Build the docker image.
4. Run the container from the image prepared above.